projects/hello_world/src/main.c

```diff
@@ -0,0 +1,46 @@
+#include "delay.h"  // For real-time delays
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"  // Software timers for scheduling future events.
+#include "wait.h"        // for the wait function
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+  bool half_second;
+} Counters;
+
+void callback(SoftTimerId timer_id, void *context) {
```

Comment:
please use `static void`, and prefix the function name with `prv_` to denote the function is private to this file/module. Could also be made more descriptive by using `prv_timer_callback`.